scripts/doc-checker.py

```diff
@@ -13,16 +13,109 @@
 from typing import List
 import sys
 
-sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
+MODIN_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+sys.path.insert(0, MODIN_PATH)
 
 # error codes that pandas test in CI
 # https://numpydoc.readthedocs.io/en/latest/validation.html#built-in-validation-checks
 NUMPYDOC_BASE_ERROR_CODES = {
-    *("GL01", "GL02", "GL03", "GL05", "GL06", "GL07", "GL09", "GL10"),
-    *("SS02", "SS04", "SS05", "PR03", "PR04", "PR05", "PR10", "RT01"),
-    *("RT04", "RT05", "SA02", "SA03"),
+    *("GL01", "GL02", "GL03", "GL05", "GL06", "GL07", "GL08", "GL09", "GL10"),
+    *("SS02", "SS03", "SS04", "SS05", "PR01", "PR02", "PR03", "PR04", "PR05"),
+    *("PR10", "RT01", "RT04", "RT05", "SA02", "SA03"),
 }
 
+""" TEST CUSTOM CHECK
+MODIN_ERROR_CODES = {
+    "TE01": "'{parameter}' description should be '{should}', found - '{found}'"
+}
+
+
+import inspect
+
+
+def get_default_args(func):
+    if not callable(func):
+        return None
+    signature = inspect.signature(func)
+    return {
+        k: v.default
+        for k, v in signature.parameters.items()
+        if v.default is not inspect.Parameter.empty
+    }
+
+
+def validate_modin_error(import_path):
+    from numpydoc.validate import Docstring
+
+    doc = Docstring(import_path)
+    errors = []
+    default_args = get_default_args(doc.obj)
+    if not default_args:
+        return None
+    for parameter in doc.doc_parameters:
+        if parameter in default_args:
+            type_line = doc.doc_parameters[parameter][0]
+            if "default " not in type_line:
+                after = type_line.split("default")[1]
+                found = "default" + after
+                should = "default [VALUE]"
+                errors.append(
+                    (
+                        "TE01",
+                        MODIN_ERROR_CODES["TE01"].format(
+                            parameter=parameter, should=should, found=found
+                        ),
+                    )
+                )
+
+    return errors
+
+
+def update_results(results, modin_errors):
+    raise NotImplementedError
+"""
+
+
+def skip_check_if_noqa(import_path):
+    """
+    Align behavior with pydocstyle.
+
+    Parameters
+    ----------
+    import_path : str
+        python-like import path
+
+    Returns
+    -------
+    bool
+        Return True if 'noqa' found.
+    """
+    import inspect
+    from numpydoc.validate import Docstring
+
+    result = False
+    doc = Docstring(import_path)
+    source = doc.method_source
+
+    noqa_str = None
+    # find last line of obj definition
+    for line in source.split("\n"):
+        if ")" in line and ":" in line.split(")", 1)[1]:
+            noqa_str = line
+            break
+
+    if noqa_str and "noqa" in noqa_str:
```

Comment:
I suggest changing how `noqa` is handled by removing the error codes mentioned in `noqa` directive unconditionally.

Reply:
What if the comment looks like this: `# noqa: E731`? This code refers to `flake8`

Reply:
This error code won't be detected by neither pydocstyle nor numpydoc, so no worries.
The format of `noqa` comment is comma-separated list of ignored checks, so I see no issues in adding new checks to ignore, and even mixing doc checks and flake checks in one `noqa` should probably be fine.

Reply:
I mean we should be handling a line like `  # noqa: E731, GL08` perfectly fine.

Reply:
Improved.